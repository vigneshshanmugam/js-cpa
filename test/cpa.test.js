jest.mock("chalk");
jest.mock('path');

const findMatches = require("../");
const { parse } = require("babylon");
const path = require("path");
const fs = require("fs");

const fixturesDir = path.join(__dirname, "fixtures");

let result,
  write = process.stdout.write,
  columns = process.stdout.columns;

beforeEach(() => {
  result = "";
  columns = 0;
  process.stdout.write = msg => {
    result += msg;
  };
});

afterEach(() => {
  process.stdout.write = write;
  process.stdout.columns = columns;
});

test("should print no matches found", () => {
  const noMatch = findMatches(fixturesDir, "nomatch.js");
  expect(result).toMatchSnapshot();
});

test("should print matches with largest subexpression in the tree", () => {
  const match = findMatches(fixturesDir, "match.js");
  expect(result).toMatchSnapshot();
});
