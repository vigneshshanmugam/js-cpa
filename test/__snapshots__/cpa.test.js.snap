// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should print matches with largest subexpression in the tree 1`] = `
"
Printing matches for - match.js

Found 2 matches

Loc - 5 : 5
function f() {}

Loc - 10 : 10
function g() {}


Found 3 matches

Loc - 9 : 11
function test() {
  return function g() {};
}

Loc - 19 : 21
function test1() {
  return function g() {};
}

Loc - 24 : 26
function test2() {
  return function g() {};
}


"
`;

exports[`should print no matches found 1`] = `
"No matches found for nomatch.js
"
`;
